README-SAMPLES.md

```diff
@@ -10,23 +10,23 @@ Exiv2 is a C++ library and a command line utility to read, write, delete and mod
 
 ### Sample Programs
 
-| Name | Purpose | More information |
-|:---  |:---  |:---              |
-| _**addmoddel**_   | Demonstrates Exiv2 library APIs to add/ modify/ delete metadata | [addmoddel](#addmoddel)<br>[https://exiv2.org/examples.html#example2](https://exiv2.org/examples.html#example2) |
-| _**exifcomment**_ | Set Exif.Photo.UserComment in an image  | [exifcomment](#exifcomment) |
-| _**exifdata**_    | Prints _**Exif**_ metadata in different formats in an image | [exifdata](#exifdata) |
-| _**exifprint**_   | Print _**Exif**_ metadata in images<br>Miscelleous other features | [exifprint](#exifprint)<br>[https://exiv2.org/examples.html#example1](https://exiv2.org/examples.html#example1) |
-| _**exifvalue**_   | Prints the value of a single _**Exif**_ tag in a file | [exifvalue](#exifvalue) |
-| _**exiv2**_       | Command line utility to read, write, delete and modify Exif, IPTC, XMP and ICC image metadata.<br>This is the primary test tool used by Team Exiv2 and can exercise almost all code in the library.  Due to the extensive capability of this utility, the APIs used are usually less obvious for casual code inspection. | [https://exiv2.org/manpage.html](https://exiv2.org/manpage.html)<br>[https://exiv2.org/sample.html](https://exiv2.org/sample.html) |
-| _**exiv2json**_   | Extracts data from image in JSON format.<br>This program also contains a parser to recursively parse Xmp metadata into vectors and objects. | [exiv2json](#exiv2json) |
-| _**geotag**_      | Reads GPX data and updates images with GPS Tags | [geotag](#geotag) |
-| _**iptceasy**_    | Demonstrates read, set or modify IPTC metadata | [https://exiv2.org/examples.html#example4](https://exiv2.org/examples.html#example4) |
-| _**iptcprint**_   | Demonstrates Exiv2 library APIs to print Iptc data | [https://exiv2.org/examples.html#example3](https://exiv2.org/examples.html#example3) |
-| _**metacopy**_    | Demonstrates copying metadata from one image to another | [metacopy](#metacopy) |
-| _**mrwthumb**_    | Sample program to extract a Minolta thumbnail from the makernote | [mrwthumb](#mrwthumb) |
-| _**xmpparse**_    | Read an XMP packet from a file, parse it and print all (known) properties. | [xmpparse](#xmpparse) |
-| _**xmpprint**_    | Read an XMP from a file, parse it and print all (known) properties.. | [xmpprint](#xmpprint) |
-| _**xmpsample**_   | Demonstrates Exiv2 library high level XMP classes | [xmpsample](#xmpsample)<br>[https://exiv2.org/examples.html#example5](https://exiv2.org/examples.html#example5) |
+| Name | Purpose | More information | Code |
+|:---  |:---  |:---              |:-- |
+| _**addmoddel**_   | Demonstrates Exiv2 library APIs to add/ modify/ delete metadata | [addmoddel](#addmoddel) | [addmoddel.cpp](https://exiv2.org/doc/addmoddel_8cpp-example.html) |
```

Comment:
Hum ... I am not sure about this change. Instead of pointing to the exiv2.org doxygen generated documentation, I would point directly to the source file in the repository. In that way, we are sure that users will see the source file corresponding to each branch.

Reply:
That's a good idea.  I didn't know that every file on GitHub is beautifully formatted in colour with line numbers (no line numbers from doxygen formatted code).

I'll chose the 'master' branch. (and not 0.27-maintenance).  These twelve samples (exifprint, geotag, etc) are seldom changed.  There's a little pull-down for different branches.

https://github.com/Exiv2/exiv2/blob/master/src/actions.cpp

Reply:
I was thinking about a relative path, so that you do not need to point to a specific branch:
https://stackoverflow.com/questions/7653483/github-relative-link-in-markdown-file

I know it works when pointing to other .md files, but I am not sure if it will work when pointing to other kind of files such as .cpp.